.github/workflows/generate-docs-for-dbt-core.yml

```diff
@@ -19,149 +19,160 @@ defaults:
         uses: actions/upload-artifact@v3
         with:
-          name: ${{needs.set_variables.outputs.index_artifact_name}}
-          path: |
-            dist/index.html
+          name: ${{ needs.set_variables.outputs.index_artifact_name }}
+          path: dist/index.html
           if-no-files-found: error
           retention-days: 15
 
+  gather_changelogs:
+    needs: [set_variables]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout dbt-docs repo
+        uses: actions/checkout@v3
+
       - name: upload changelog artifacts
         uses: actions/upload-artifact@v3
         with:
-          name: ${{needs.set_variables.outputs.changie_artifact_name}}
+          name: ${{ needs.set_variables.outputs.changie_artifact_name }}
           path: |
-            .changes/unreleaded/*.yml
+            .changes/unreleased/*.yaml
           if-no-files-found: error
           retention-days: 15
 
-  # create branch on dbt-core
-  create_dbt_core_branch:
+  commit_dbt_core:
     needs: [set_variables, build_index]
     runs-on: ubuntu-latest
 
     steps:
-      - name: Check out repository code
+      - name: Check out dbt-core main branch
         uses: actions/checkout@v3
         with:
           repository: dbt-labs/dbt-core
           ref: main
-          # token: PAT?
+          token: ${{ secrets.FISHTOWN_BOT_PAT }}
  
       - name: Create branch
         run: |
-          git checkout -b ${{needs.set_variables.outputs.branch_name}}
-          git push -u origin ${{needs.set_variables.outputs.branch_name}}
+          git checkout -b ${{ needs.set_variables.outputs.branch_name }}
+          git push -u origin ${{ needs.set_variables.outputs.branch_name }}
       
-      - name: Check current branch
+      - name: Print branch status
         run: |
           git status
 
-  commit_dbt_core:
-    needs: [set_variables, create_dbt_core_branch]
-    runs-on: ubuntu-latest
-
-    steps:
-
-      - name: Check out the dbt-core branch
-        uses: actions/checkout@v3
+      - name: Download index artifact
+        uses: actions/download-artifact@v3
         with:
-          repository: dbt-labs/dbt-core
-          ref: ${{ needs.set_variables.outputs.branch_name }}
-          # token: PAT?
-
-      - uses: actions/download-artifact@v3
-        with:
-          name: ${{needs.set_variables.outputs.index_artifact_name}}
+          name: ${{ needs.set_variables.outputs.index_artifact_name }}
           path: "core/dbt/include"
       
-      - uses: actions/download-artifact@v3
+      - name: Download changelog artifact(s)
+        uses: actions/download-artifact@v3
         with:
-          name: ${{needs.set_variables.outputs.changie_artifact_name}}
+          name: ${{ needs.set_variables.outputs.changie_artifact_name }}
           path: ".changes/unreleased"
 
       - name: Print branch status
         run: |
           git status
 
+      # note that if nothing has changed with the repo, the index file won't be
```

Comment:
So just the changelog files will show up with the work that was done but the missing index file won't reflect that?

Reply:
Honestly, I can't think of a reason this would happen.  It happened while I was testing (there were no changes to the index file and it took me a minute to figure out why the index file wasn't showing up in the PR).  I could add a warning or error if this happens so it's clearer than a comment in the code.  

If the index file doesn't change, I don't know why we would want to have changelogs published for `dbt-docs` in `dbt-core`.

Reply:
Ok yeah that's what I was thinking as well (changelogs exist but no index file change). I think this is fine for now and we can always add a message if we do encounter this